services/py-api/src/database/transaction_manager.py

```diff
@@ -55,22 +57,23 @@ async def _retry_tx(func: Callable[..., Awaitable[Result]], *args: Any, **kwargs
         """
         max_retries = 3
         delay = 1  # initial delay in seconds
+        session_id = session.session_id["id"].hex()
 
         # range is exclusive that's why we do max_retries + 1
         for retry in range(1, max_retries + 1):
             try:
-                return await func(*args, **kwargs)
+                return await func(*args, session=session, **kwargs)
             except PyMongoError as exc:
                 if exc.has_error_label("TransientTransactionError"):
-                    LOG.debug("Retrying transaction retry {}".format(retry))
+                    LOG.debug(f"Retrying transaction retry {retry}", session_id=session_id)
                     await sleep(delay)
                     delay *= 2  # exponential backoff
                     continue
 
                 # If the exception it's a non-retryable error re-raise it in order to be caught on an upper level
                 raise exc
 
-        raise PyMongoError("Transaction failed after maximum retries")
+        raise PyMongoError(f"ID:[{session_id}] Transaction failed after maximum retries")
```

Comment:
```python
raise PyMongoError(f"Transaction with ID:[{session_id}] failed after maximum retries")
```

Reply:
ok

Reply:
Fixed in 62ecace76fd209bffa772b0238adedf16bd38dc2